ops/_private/harness.py

```diff
@@ -3486,10 +3486,8 @@ def list_files(
             raise self._api_error(404, f'stat {path}: no such file or directory')
         files = [file_path]
         if not itself:
-            try:
+            with contextlib.suppress(NotADirectoryError):
                 files = [file_path / file for file in os.listdir(file_path)]
-            except NotADirectoryError:
-                pass
```

Comment:
I'm not a fan of `contextlib.suppress`. I don't see how its use improves code at all - the reader needs to know that this exists and what it does, there's no improvement in indentation, it's two lines briefer but *longer* in terms of characters, and I don't think it's any more understandable (although that's subjective of course).

My preference would be to leave this as-is.